### ISSUE TITLE
RedundantCondition seen for functions that return a resource or false.

### ISSUE THREAD

**Issue by TysonAndre** [vimeo/psalm#353]
Technically, this also includes fwrite, fread, etc., which returns int|false, string|false.

CallMap.php's signatures were originally written to assume success (`resource` here) to reduce false positives. This would be nice to fix. However, continuing to not emit issues for the possibility of false for functions such as fopen, fsockopen, etc. would still be desirable

```php
// assume host, port, buffer are defined.
$fp = fsockopen("udp://$host", $port, $errno, $errstr);
if (!$fp) { return; }  // This snippet emits RedundantCondition
fwrite($fp, $buffer);
fclose($fp);
```

Related: openssl_decrypt can return non-string on error, imagecreate* can return false instead of resource, etc.

**Comment by muglug** [comment]
Yup, I'll get to those at some point. There's also the `PossiblyFalseArgument` issue which can be `@psalm-ignore`d in those contexts (if you don't want to write error handling code.
